src/onnx/parse_topk.cpp

```diff
@@ -0,0 +1,58 @@
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/make_op.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_topk : op_parser<parse_topk>
+{
+    std::vector<op_desc> operators() const { return {{"TopK"}}; }
+
+    std::vector<instruction_ref> parse(const op_desc& /*opd*/,
+                                       const onnx_parser& parser,
+                                       onnx_parser::node_info info,
+                                       std::vector<instruction_ref> args) const
+    {
+        int64_t k = 0;
+        if(args.size() == 2)
+        {
+            auto arg_k = args.at(1)->eval();
+            check_arg_empty(arg_k, "PARSE_TopK: k input must be constant");
+            k = arg_k.at<int>();
+        }
+        else if(contains(info.attributes, "k"))
+        {
+            k = info.attributes.at("k").i();
+        }
+
+        bool largest = true;
+        if(contains(info.attributes, "largest"))
+        {
+            largest = static_cast<bool>(info.attributes.at("largest").i());
+        }
+
+        int64_t axis = -1;
+        if(contains(info.attributes, "axis"))
+        {
+            axis = parser.parse_value(info.attributes.at("axis")).at<int>();
+        }
+
+        auto topk_ret = info.add_instruction(
+            make_op("topk", {{"k", k}, {"axis", axis}, {"largest", largest}}), args.at(0));
+        auto out_s = topk_ret->get_shape();
```

Comment:
when will topk not return a tuple type?

Reply:
That is specified by the onnx spec, it has two outputs, so we have to use the tuple type.

Reply:
We can return multiple outputs when parsing onnx, so we dont always need to use a tuple.

Reply:
If we don't always need to use a tuple, then I'm not sure why we need the assert on L47.

Reply:
The problem is, when we compute the output shape of this operator, we do not know whether one or two outputs are needed, so we have to provide two outputs, and the output shape is a tuple. This is from the compute_shape() method, then in the compute() method, based on the output_shape computed from the compute_shape() method, we return an argument with the tuple_type, this is required by our design since we cannot change the output shape computed from the compute_shape() method.

I know for the RNN Operators, we have multiple outputs, but we did not use the tuple type, the reason for that is, we compute a first output, which contains information for the second output, then we insert another operator (rnn_last_output) to retrieve the second output from the first output. This is a special case so we can avoid the tuple type. 

If we understand in another way, for topk, we get the tuple output argument, then we insert the get_tuple_elem operator, this is very similar as what we did for the RNN operator. With our current design and architecture, I do not know how to avoid the tuple type. Of course, we can come up with anther design to not use the tuple.

Reply:
I was mainly bringing up that the assert would always pass? Is it necessary?

Reply:
I just removed it.